plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,463 @@
+		"datainvalidates",
+		"attrinvalidates",
+		"vfsopen",
+		"vfslookup",
+		"vfsaccess",
+		"vfsupdatepage",
+		"vfsreadpage",
+		"vfsreadpages",
+		"vfswritepage",
+		"vfswritepages",
+		"vfsgetdents",
+		"vfssetattr",
+		"vfsflush",
+		"vfsfsync",
+		"vfslock",
+		"vfsrelease",
+		"congestionwait",
+		"setattrtrunc",
+		"extendwrite",
+		"sillyrenames",
+		"shortreads",
+		"shortwrites",
+		"delay",
+		"pnfsreads",
+		"pnfswrites",
+	}
+
+	var bytesFields = []string{
+		"normalreadbytes",
+		"normalwritebytes",
+		"directreadbytes",
+		"directwritebytes",
+		"serverreadbytes",
+		"serverwritebytes",
+		"readpages",
+		"writepages",
+	}
+
+	var xprtudpFields = []string{
+		"bind_count",
+		"rpcsends",
+		"rpcreceives",
+		"badxids",
+		"inflightsends",
+		"backlogutil",
+	}
+
+	var xprttcpFields = []string{
+		"bind_count",
+		"connect_count",
+		"connect_time",
+		"idle_time",
+		"rpcsends",
+		"rpcreceives",
+		"badxids",
+		"inflightsends",
+		"backlogutil",
+	}
+
+	var nfsopFields = []string{
+		"ops",
+		"trans",
+		"timeouts",
+		"bytes_sent",
+		"bytes_recv",
+		"queue_time",
+		"response_time",
+		"total_time",
+		"errors",
+	}
+
+	var fields = make(map[string]interface{})
+
+	switch first {
+	case "READ", "WRITE":
+		fields["ops"] = nline[0]
+		fields["retrans"] = nline[1] - nline[0]
+		fields["bytes"] = nline[3] + nline[4]
+		fields["rtt"] = nline[6]
+		fields["exe"] = nline[7]
+		tags["operation"] = first
+		acc.AddFields("nfsstat", fields, tags)
+	}
+
+	if fullstat {
+		switch first {
+		case "events":
+			if len(nline) >= len(eventsFields) {
+				for i, t := range eventsFields {
+					fields[t] = nline[i]
+				}
+				acc.AddFields("nfs_events", fields, tags)
+			}
+
+		case "bytes":
+			if len(nline) >= len(bytesFields) {
+				for i, t := range bytesFields {
+					fields[t] = nline[i]
+				}
+				acc.AddFields("nfs_bytes", fields, tags)
+			}
+
+		case "xprt":
+			if len(line) > 1 {
+				switch line[1] {
+				case "tcp":
+					if len(nline)+2 >= len(xprttcpFields) {
+						for i, t := range xprttcpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_tcp", fields, tags)
+					}
+				case "udp":
+					if len(nline)+2 >= len(xprtudpFields) {
+						for i, t := range xprtudpFields {
+							fields[t] = nline[i+2]
+						}
+						acc.AddFields("nfs_xprt_udp", fields, tags)
+					}
+				}
+			}
+		}
+
+		if (version == "3" && n.nfs3Ops[first]) || (version == "4" && n.nfs4Ops[first]) {
+			tags["operation"] = first
+			if len(nline) <= len(nfsopFields) {
+				for i, t := range nline {
+					fields[nfsopFields[i]] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
+
+	}
+
+	return nil
+}
+
+func (n *NFSClient) processText(scanner *bufio.Scanner, acc telegraf.Accumulator) error {
+	var mount string
+	var version string
+	var export string
+	var skip bool
+
+	for scanner.Scan() {
+		line := strings.Fields(scanner.Text())
+
+		line_len := len(line)
+
+		if line_len == 0 {
+			continue
+		}
+
```

Comment:
```suggestion
		skip = false
```
Otherwise `skip` might still contain the value of the previous scan (i.e. undefined) if you don't enter the first `if` and `len(n.IncludeMounts) == 0`.

Reply:
I think there was a reason for this early on, which may no longer apply.  I'll rig some new test cases and look later on.

Reply:
Thanks, it's just a bit strange. :-) So even if it is what you intended please put a comment explaining it to others.

Reply:
I believe the idea is to use a "carryover" value from parsing the previous line, so as to skip uneeded ExcludeMount checks.  I've tested it both ways, and there's no difference in behavior.  But explicitly initializing the value is good.  

The bigger fix is a different bug introduced by calling `return` too soon when checking exclude blocks; it should be a `break` instead, since `return` will halt processing of the rest of the data.

Test cases:

No includes or excludes:
```
# include_mounts =  = ['/A', 'B$' ]
# exclude_mounts = ['/B', 'C' ]
--> All mounts are included.
```

Only includes:
```
include_mounts =  = ['/A', 'B$' ]
# exclude_mounts = ['/B', 'C' ]
--> Only mounts /A and /B are included.
```
Only excludes:
```
# include_mounts =  = ['/A', 'B$' ]
exclude_mounts = ['/B', 'C' ]
--> Mounts /A and /D are included.
```
Both includes and excludes:
```
include_mounts =  = ['/A', 'B$' ]
exclude_mounts = ['/B', 'C' ]
--> Only mount /A is included
```